src/App.js

```diff
@@ -14,11 +16,114 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.PureComponent {
+  state = {
+    hideButton: false,
+    showList: true,
+    isReversed: false,
+    sortBy: '',
+    value: '1',
+  }
+
+  handleChange = (event) => {
+    this.setState({ value: event.target.value });
+  }
+
+  startButton = () => {
+    this.setState(prevState => ({
+      hideButton: !prevState.hideButton,
+      showList: false,
+    }));
+  }
+
+  reverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  }
+
+  sortAlph = () => {
+    this.setState({ sortBy: 'alphabet' });
+  }
+
+  sortLength = () => {
+    this.setState({ sortBy: 'length' });
+  }
+
+  reset = () => {
+    this.setState({
+      sortBy: '', isReversed: false, value: '1',
+    });
+  }
+
+  render() {
+    const { isReversed, sortBy, showList, hideButton, value } = this.state;
+
+    return (
+      <div className="App box has-text-centered">
+        <h1 className="title is-1">Goods</h1>
+        <GoodsList
+          showList={showList}
+          goods={goodsFromServer}
+          isReversed={isReversed}
+          sortBy={sortBy}
+          value={value}
+        />
```

Comment:
You should do all the operations (sorting, reversing etc.) on this level (in the parent element) and pass only preparedGoods to 'GoodsList' component, which should only render the list with already prepared (sorted or reversed or whatever) goods